CRM/Price/BAO/PriceSet.php

```diff
@@ -1513,4 +1513,16 @@ public static function flushPriceSets(): void {
     }
   }
 
+  /**
+   * Whitelist of possible values for the entity_table field in PriceSetEntity
+   *
+   * @return array
+   */
+  public static function entityTables(): array {
+    return [
+      'civicrm_contribution_page' => ts('Contribution Page'),
+      'civicrm_event' => ts('Event'),
```

Comment:
Are we sure this is the complete list of entities that can be extended?

Reply:
I'm not 100% sure but looking through the code I'm about 95% sure. It shouldn't actually break anything if it's not the full list though, just that when using the UI you won't be able to select other entities.
But as far as I can tell the "PriceSetEntity" was only ever used to link event registration and contributionpages to their respective pricesets. I don't think there is anything else in Civi that uses pricesets?

Reply:
I think this is OK - It's inline with what we have done on `LineItem` - I feel like @sunilpawar might have hit some issue wanting pledge to work there - but I see we didn't alter it. There is also some stuff by JMA (@seamuslee001 @monishdeb ) that does some stuff with grants - but I don't think it would be affected